type-overrides/src/lib.rs

```diff
@@ -0,0 +1,52 @@
+///
+/// This lib contains both standard imports and imports shuttle.
+/// Shuttle is a Rust crate that facilitates multithreaded testing. It has its own scheduler
+/// and can efficiently detect bugs in concurrent code. The downside is that we need to replace
+/// all imports by those from Shuttle.
+///
+/// Instead of importing from std, rand, and so on, import the following from solana-type-override,
+/// and include the 'shuttle-test' feature in your crate to use shuttle.
+pub mod executor {
+    #[cfg(not(feature = "shuttle-test"))]
+    pub use futures::executor::*;
+    #[cfg(feature = "shuttle-test")]
+    pub use shuttle::future::*;
+}
+
+pub mod hint {
+    #[cfg(feature = "shuttle-test")]
+    pub use shuttle::hint::*;
+    #[cfg(not(feature = "shuttle-test"))]
+    pub use std::hint::*;
+}
+
+pub mod lazy_static {
+    #[cfg(not(feature = "shuttle-test"))]
+    pub use lazy_static::*;
+    #[cfg(feature = "shuttle-test")]
+    pub use shuttle::lazy_static::*;
+}
+
+pub mod rand {
+    pub use rand::{
```

Comment:
why not `rand::*`?

Reply:
Can Rust simply override the import when I include the same names from shuttle? I thought it would lead to a name conflict.

Reply:
I'm pretty sure it overrides. Only one way to know! 😋

Reply:
I tested it here. It worked. Let me just rebase the PR to fix the conflicts.